src/api/MainSolver.cc

```diff
@@ -333,14 +334,23 @@ sstat MainSolver::check() {
         StopWatch sw(query_timer);
     }
     if (isLastFrameUnsat()) { return s_False; }
-    sstat rval = simplifyFormulas();
+    sstat rval;
+    try {
+        rval = simplifyFormulas();
+    } catch (NonLinException const & error) {
+        reasonUnknown = error.what();
+        return s_Undef;
+    }
 
     if (config.dump_query()) printCurrentAssertionsAsQuery();
 
     if (rval == s_Undef) {
         try {
             rval = solve();
-        } catch (std::overflow_error const & error) { rval = s_Error; }
+        } catch (std::overflow_error const & error) { rval = s_Error; } catch (NonLinException const & error) {
```

Comment:
```suggestion
        } catch (std::overflow_error const & error) { 
            rval = s_Error;
        } catch (NonLinException const & error) {
```
Is this some formatter thing? The two catch clauses should not be on the same line.

Reply:
It seems like a bug in `clang-format`. We use `LLVM` style which seems to imply `BreakBeforeBraces: Attach` (see [here](https://releases.llvm.org/19.1.0/tools/clang/docs/ClangFormatStyleOptions.html)). All other styles correctly wrap each `catch`, but not in the case of `Attach` style.

Reply:
It is a formatter thing unfortunately, and clang-format throws an error otherwise :((